utils/staking-miner/src/monitor.rs

```diff
@@ -54,17 +63,63 @@ async fn ensure_no_previous_solution<
 	T: EPM::Config + frame_system::Config<AccountId = AccountId>,
 	B: BlockT,
 >(
-	ext: &mut Ext,
+	client: &WsClient,
+	at: B::Hash,
 	us: &AccountId,
 ) -> Result<(), Error<T>> {
-	use EPM::signed::SignedSubmissions;
-	ext.execute_with(|| {
-		if <SignedSubmissions<T>>::get().iter().any(|ss| &ss.who == us) {
-			Err(Error::AlreadySubmitted)
-		} else {
-			Ok(())
+	let indices_key = storage_value(EPM_MODULE_PREFIX, b"SignedSubmissionIndices");
```

Comment:
Also, as discussed, you can make `SignedSubmissionIndices` public.

Reply:
I had some troubles to actually use the `StorageValue trait` here because of some trait bounds that are missing from the `Runtime`.

```bash
let xx = StorageValue::<EPM::SignedSubmissions::<Runtime>>::hashed_key();

    = note: required because of the requirements on the impl of `FullEncode` for `SignedSubmissions<kusama_runtime::Runtime>`
    = note: required because of the requirements on the impl of `FullCodec` for `SignedSubmissions<kusama_runtime::Runtime>`
note: required by a bound in `hashed_key`
```